### PR TITLE
add hostmetrics receiver components to otel docs

### PR DESCRIPTION
**Pull request by frzifus**
Version(s):

Issue:
- https://issues.redhat.com/browse/TRACING-3977
- https://issues.redhat.com/browse/TRACING-3962
- https://issues.redhat.com/browse/TRACING-4076
- https://issues.redhat.com/browse/TRACING-4073

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:
Removed the hostmetrics part as discussed.

### CODE REVIEW

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
 <10> The path to the client's private key file that is used with the client certificate if needed for authentication.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
+
+The File Storage extension can persist state to the local file system.
+The extension requires read and write access to a directory. A default directory can be used, but it must already exist in order for the extension to operate.
+
+The OAuth2Client extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:947]: @frzifus, is  `OAuth2Client` mentioned here intentionally or is this just a leftover from the copy-and-paste steps when preparing this draft?

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
...
+        traces:
+          receivers: [otlp]
+          exporters: [otlp]
+----
+<1> Specifies the directory in which the telemetry data is stored.
+<2> Specifies the a timeout to open the storedfiles.
+<3> Starts the compaction when the collector starts. If omitted, the default `+false+` is used.
+<4> Specifies the directory in which compactor stores the telemetry data is stored.
+<5> Defines maximum size of the compaction transaction. A value of zero will ignore transaction sizes. If omitted, the default `+65536+` is used.
+<6> When set, will force the database to perform an fsync after each write. This helps to ensure database integretity if there is an interruption to the database process, but at the cost of performance.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:981]: @frzifus, what is `fsync`? Can you please define and explain it to me here in a comment?

File: modules/otel-collector-components.adoc [hunk 3]
```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
...
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
+
+The File Storage extension can persist state to the local file system.
+The extension requires read and write access to a directory. A default directory can be used, but it must already exist in order for the extension to operate.
+
+The OAuth2Client extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
+This extension supports traces, metrics, and logs.
 
+.OpenTelemetry Collector custom resource with configured FileStorage extension to persist as OTLP sendig queue.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:950]: @frzifus, could you please reply in this thread to explain this phrase to me the way you understand it: `to persist as OTLP sendig queue` ?

File: modules/otel-collector-components.adoc [hunk 4]
```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
 <10> The path to the client's private key file that is used with the client certificate if needed for authentication.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:942]: @frzifus, is this extension in Technology Preview for 3.2?

File: modules/otel-collector-components.adoc [hunk 5]
```diff
@@ -82,6 +82,39 @@ The Jaeger receiver ingests traces in the Jaeger formats.
 <4> The Jaeger Thrift Binary endpoint. If omitted, the default `+0.0.0.0:6832+` is used.
 <5> The  server-side TLS configuration. See the OTLP receiver configuration section for more details.
 
+[id="k8sobjectsreceiver-receiver_{context}"]
+=== K8S Objects Receiver
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:86]: @frzifus, is this receiver in Technology Preview for 3.2?

File: modules/otel-collector-components.adoc (hunk 3 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:950]: The OTLP exporter has a sending queue where all messages that should be published get "parked" until they are successfully transmitted.
By default this is in memory. If you restart the collector, this data will be lost.

The file storage extension implements the interface of this queue and writes the messages to disk. If you restart the collector, all non transmitted messages should still exist.

File: modules/otel-collector-components.adoc (hunk 1 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:947]: Ups, thats a leftover from copy pasta.

```suggestion
The FileStorage extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
```

File: modules/otel-collector-components.adoc (hunk 2 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:981]: `fsync` is a call that tells the operating system to write changed files (which are potentially still in memory) to the underlying storage medium (e.g. a hard drive disk).

https://man7.org/linux/man-pages/man2/fsync.2.html

File: modules/otel-collector-components.adoc [hunk 6]
```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
...
+      pipelines:
+        traces:
+          receivers: [otlp]
+          exporters: [otlp]
+----
+<1> Specifies the directory in which the telemetry data is stored.
+<2> Specifies the a timeout to open the storedfiles.
+<3> Starts the compaction when the collector starts. If omitted, the default `+false+` is used.
+<4> Specifies the directory in which compactor stores the telemetry data is stored.
+<5> Defines maximum size of the compaction transaction. A value of zero will ignore transaction sizes. If omitted, the default `+65536+` is used.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:980]: @frzifus, I think we will need to mention the units for this field. Can you find out?

File: modules/otel-collector-components.adoc (hunk 5 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:86]: yes

File: modules/otel-collector-components.adoc (hunk 4 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:942]: yes

File: modules/otel-collector-components.adoc (hunk 6 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:980]: The unit is bytes. In [Compact](https://github.com/etcd-io/bbolt/blob/51926613987d57aa9df1f1bfcce0467598932e90/compact.go#L8-L50) they have a check if the addded keys+values exceed the max_transaction_size. If its the case, they commit the current transaction and create a new one.

File: modules/otel-collector-components.adoc (hunk 4 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:942]: Updated.

File: modules/otel-collector-components.adoc (hunk 5 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:86]: Updated.

File: modules/otel-collector-components.adoc [hunk 7]
```diff
@@ -684,6 +684,42 @@ The Debug exporter prints traces and metrics to the standard output.
 ----
 <1> Verbosity of the debug export: `detailed` or `normal` or `basic`. When set to `detailed`, pipeline data is verbosely logged. Defaults to `normal`.
 
+[id="otlp-http-exporter_{context}"]
+=== OTLP HTTP exporter
+
+This is an exporter that will consistently export spans, metrics and logs depending on the routing_key configured.
+
+.OpenTelemetry Collector custom resource with an enabled OTLP exporter
+[source,yaml]
```
Comment (max-cx) [inline modules/otel-collector-components.adoc]: We already have the OTLP HTTP exporter: https://docs.openshift.com/container-platform/4.14/observability/otel/otel-configuration-of-otel-collector.html#otlp-http-exporter_otel-configuration-of-otel-collector
@frzifus, :slightly_smiling_face: are you updating the existing feature or introducing another, new feature? If it's a new feature, then what is its name and is it an exporter or extension or etc.?

Comment (frzifus) [reply modules/otel-collector-components.adoc]: oh thats my fault. Its the LB exporter.

File: modules/otel-collector-components.adoc [hunk 8]
```diff
@@ -869,7 +938,47 @@ This extension supports traces, metrics, and logs.
...
+        compaction:
+          on_start: true # <3>
+          directory: /tmp/ # <4>
+          max_transaction_size: 65_536 # <5>
+        fsync: false # <6>
 
+    exporters:
+      otlp:
+        sending_queue:
+          storage: file_storage/all_settings # <7>
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:967]: Should we add a callout for `<7>` below? Or is this just a dangling typo?

Comment (frzifus) [reply modules/otel-collector-components.adoc:967]: Oh, we should defiantly add something.
```
7: Register the file_storage/all_settings component in the otlp exporter that should be buffered on disk.
```

File: modules/otel-collector-components.adoc [hunk 9]
```diff
@@ -869,7 +938,47 @@ This extension supports traces, metrics, and logs.
...
+
+    service:
+      extensions: [file_storage/all_settings]
```
Comment (frzifus) [inline modules/otel-collector-components.adoc:970]: We should also mention that the extension needs to be added here. e.g. <8>:
```
8: register the file_storage extension to be started by the opentelemetry-collector.
```

**Comment by max-cx** [comment]
Thank you, Bene. I'm closing this PR. I'll add the two last-minute callouts in a separate Jira ticket: https://issues.redhat.com/browse/OBSDOCS-1097

File: modules/otel-collector-components.adoc (hunk 8 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:967]: @frzifus, :slightly_smiling_face:  what do you mean by "Register" here? Could you please explain it to me in some other words?

File: modules/otel-collector-components.adoc (hunk 9 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:970]: @frzifus, :slightly_smiling_face:  what do you mean by "Register" here? Could you please explain it to me in some other words?

File: modules/otel-collector-components.adoc (hunk 8 above)
Comment (frzifus) [reply modules/otel-collector-components.adoc:967]: To be able to use the `file_storage/all_settinghs` extension here, you need to create an instance or start it internally.  Thats done by listing it in the `servcie::extensions` section. Does this make sense?

Comment (max-cx) [reply modules/otel-collector-components.adoc:967]: @frzifus, what do you mean by "buffered on disk" in your suggestion below?

`7: Register the file_storage/all_settings component in the otlp exporter that should be buffered on disk.`

Comment (frzifus) [reply modules/otel-collector-components.adoc:967]: Usually the process will keep all measurements in an in-memory queue. If the process restarts, that data will be lost. Using the `file_storage` extension this data will be written to disk. If the collector restarts, the measurements will still exist and can still be transmitted to a remote backend.

Comment (max-cx) [reply modules/otel-collector-components.adoc:967]: @frzifus, :sweat_smile: what is "disk"? I know this sounds like a very dumb question, but specifically? Is it a Persistent Volume?

I see this sentence in the extension description: "This extension can persist the state to the local file system." Is this what you mean?

Comment (frzifus) [reply modules/otel-collector-components.adoc:967]: yes :+1:

**Comment by max-cx** [comment]
I've opened https://github.com/openshift/openshift-docs/pull/79739 to document the initially left out callouts 7 and 8.

**Pull request closed without merging**